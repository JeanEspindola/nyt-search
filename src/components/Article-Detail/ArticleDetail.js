/*
* Article Detail Presentation Component
*
* Renders the detailed list. Using moment library to format dates.
* Using Link to make sure to call parant route.
*
* */
import React, { Component } from 'react';
import { Row, Glyphicon } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import Moment from 'moment';
import './ArticleDetail.css';

class ArticleDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {
      article: this.props.location.article,
    };
    Moment.locale('de');
  }

  render() {
    const { article } = this.state;
    if (article === undefined) {
      return <p>No details to show.</p>;
    }

    return (
      <div className="Detail">
        <Row className="Detail__Return">
          <Link to="/" >
            <Glyphicon glyph="chevron-left" />
            Go to Results Page
          </Link>
        </Row>
        <Row className="Detail__Title">
          <h3>{article.headline.main}</h3>
        </Row>
        <Row className="Detail__Date">
          {Moment(article.pub_date).format('DD.MM.YYYY')}
        </Row>
        <Row className="Detail__Text">
          {article.snippet}
        </Row>
        <Row className="Detail__Link">
          <a target="_blank" href={article.web_url}>
            Read the Full Article
          </a>
        </Row>
      </div>
    );
  }
}

export default ArticleDetail;
