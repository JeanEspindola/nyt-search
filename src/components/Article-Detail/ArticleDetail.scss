.Detail {
  max-width: 700px;
  margin: 0 auto;

  .Detail__Return {
    margin-top: 15px;
  }

  .Detail__Title {
    margin-top: 20px;
  }

  .Detail__Date {
    margin-top: 10px;
    color: grey;
  }

  .Detail__Text {
    margin-top: 20px;
  }

  .Detail__Link {
    margin-top: 50px;
  }
}