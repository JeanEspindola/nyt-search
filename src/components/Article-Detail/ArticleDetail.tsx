/*
 * Article Detail Presentation Component
 *
 * Renders the detailed list. Using moment library to format dates.
 * Using Link to make sure to call parent route.
 *
 * */
import { useLocation } from 'react-router'
import { Row } from 'react-bootstrap'
import * as Icon from 'react-bootstrap-icons'
import { Link } from 'react-router-dom'
import Moment from 'moment'
import styles from './ArticleDetail.module.scss'
import { FormattedMessage } from 'react-intl'

const isArticleEmpty = (article: Article) =>
  article === null || article === undefined || Object.keys(article).length === 0

interface Article {
  headline: {
    main: string
  }
  pub_date: string
  snippet: string
  web_url: string
}

const ArticleDetail = () => {
  Moment.locale('de')

  const article = useLocation<{ article: Article }>().state?.article ?? ''

  return (
    <>
      {isArticleEmpty(article) ? (
        <p>
          <FormattedMessage id={'noDetails'} />
        </p>
      ) : (
        <div className={styles.container}>
          <Row className={styles.return}>
            <Link to="/">
              <Icon.ChevronLeft />
              <FormattedMessage id={'goToResults'} />
            </Link>
          </Row>
          <Row className={styles.title}>
            <h3>{article.headline.main}</h3>
          </Row>
          <Row className={styles.date}>
            {Moment(article.pub_date).format('DD.MM.YYYY')}
          </Row>
          <Row className={styles.text}>{article.snippet}</Row>
          <Row className={styles.link}>
            <a target="_blank" href={article.web_url} rel="noopener noreferrer">
              <FormattedMessage id={'readArticle'} />
            </a>
          </Row>
        </div>
      )}
    </>
  )
}

export default ArticleDetail
