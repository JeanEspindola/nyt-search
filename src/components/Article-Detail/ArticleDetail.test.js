import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ArticleDetail from './ArticleDetail';

describe('ArticleDetail', () => {
  let Component;
  let location;
  const article = {
    headline: {
      main: 'test',
    },
    pub_date: '2018-01-14',
    snippet: 'test test test',
    web_url: 'www.test.com',
  };

  it('renders ArticleDetail without crashing', () => {
    location = {
      article,
    };
    Component = shallow(<ArticleDetail location={location} />);
    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });

  it('renders ArticleDetail with no props', () => {
    const text = 'No details to show.';
    location = {
      article: undefined,
    };

    Component = shallow(<ArticleDetail location={location} />);
    expect(Component.find('p').text()).toBe(text);
  });
});

