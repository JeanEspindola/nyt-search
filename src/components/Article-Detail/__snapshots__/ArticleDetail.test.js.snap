// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetail renders ArticleDetail without crashing 1`] = `
<div
  className="Detail"
>
  <Row
    bsClass="row"
    className="Detail__Return"
    componentClass="div"
  >
    <Link
      replace={false}
      to="/"
    >
      <Glyphicon
        bsClass="glyphicon"
        glyph="chevron-left"
      />
      Go to Results Page
    </Link>
  </Row>
  <Row
    bsClass="row"
    className="Detail__Title"
    componentClass="div"
  >
    <h3>
      test
    </h3>
  </Row>
  <Row
    bsClass="row"
    className="Detail__Date"
    componentClass="div"
  >
    14.01.2018
  </Row>
  <Row
    bsClass="row"
    className="Detail__Text"
    componentClass="div"
  >
    test test test
  </Row>
  <Row
    bsClass="row"
    className="Detail__Link"
    componentClass="div"
  >
    <a
      href="www.test.com"
      target="_blank"
    >
      Read the Full Article
    </a>
  </Row>
</div>
`;
